### PR TITLE
feat: add `@kintone/profile-loader`

### PR DESCRIPTION
**Pull request by koba04**
## TODO

- [x] Support `KINTONE_PROFILE`
- [x] Rename from `KINTONE_CONFIG_PATH` and `KINTONE_CREDENTIALS_PATH` to `KINTONE_CONFIG_FILE` and `KINTONE_CREDENTIALS_FILE`

## Why

Currently, we don't provide a way to manage settings like `baseUrl`, `username`, `password`, and `apiToken`, so developers manage the settings on their own.
In addition to that, I heard a feature request to switch settings between environments, which are `dev`, `stage`, and `production`.

So I'd like to introduce a new package called `@kintone/profile-loader`.

**This is in an experimental stage, so I don't have a plan to publish it yet, which is only used in packages in `kintone/js-sdk`.**

## What

See [the rendered description](https://github.com/koba04/js-sdk/blob/profile-loader/packages/profile-loader/README.md)

For library authors, you can use this as an npm package.
For users of the libraries, you have to configure kintone settings through environmental variables or `$HOME/.kintone/{config, credentials}` files.

## How to test

You can try this at `examples/rest-api-client-demo`.
To run `rest-api-client-demo` scripts, you have to set environmental variables or put `$HOME/.kintone/{config, credentials}` files to set `baseUrl`, `username`,  `password`, `apiToken`.

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: packages/profile-loader/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# `@kintone/profile-loader`
+
+A loader to retrive Kintone settings like `username`, `password`, `baseUrl`, and whatever you want.
```
Comment (shisama) [inline packages/profile-loader/README.md:3]:
```suggestion
A loader to retrieve Kintone settings like `username`, `password`, `baseUrl`, and whatever you want.
```

File: packages/profile-loader/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  credentials: credentialsFilePath,
+});
+```
+
+## Settings
+
+### Config file
+
+You can get the following settings from the function.
+each setting corresponds the environment variable.
```
Comment (shisama) [inline packages/profile-loader/README.md:26]:
```suggestion
Each setting corresponds the environment variable.
```

File: packages/profile-loader/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+[staging]
+baseUrl = "https://staging-example.kintone.com"
+# Custom settings
+debug = false
+```
+
+### Credentialsfile
+
+You can get the following settings from the function.
+each setting corresponds the environment variable.
```
Comment (shisama) [inline packages/profile-loader/README.md:54]:
```suggestion
Each setting corresponds the environment variable.
```

**Comment by koba04** [comment]
I'll rename `KINTONE_CONFIG_PATH` and `KINTONE_CREDENTIALS_PATH` to `KINTONE_CONFIG_FILE` and `KINTONE_CREDENTIALS_FILE`

**Comment by koba04** [comment]
I've renamed `KINTONE_CONFIG_PATH` and `KINTONE_CREDENTIALS_PATH` to `KINTONE_CONFIG_FILE` and `KINTONE_CREDENTIALS_FILE`.

File: packages/profile-loader/package.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/kintone/js-sdk.git",
+    "directory": "packages/profile-loader"
+  },
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/kintone/js-sdk/issues"
+  },
+  "homepage": "https://github.com/kintone/js-sdk/tree/master/packages/rest-api-client#readme",
```
Comment (koba04) [inline packages/profile-loader/package.json:31]:
```suggestion
  "homepage": "https://github.com/kintone/js-sdk/tree/master/packages/profile-loader#readme",
```

**Comment by chick-p** [comment]
👍

**Pull request merged**